src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,120 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetTimePeriodForIvector(const Nnet &nnet) {
+  std::vector<std::string> name;
+  name.push_back("ivector");
```

Comment:
add a comment here saying that you are assuming that the name of the node with ivector is "ivector". This assumption is made in all binaries.